go/vt/vtgate/gateway/discoverygateway.go

```diff
@@ -273,13 +273,51 @@ func (dg *discoveryGateway) withRetry(ctx context.Context, target *querypb.Targe
 	return NewShardError(err, target, tabletLastUsed, inTransaction)
 }
 
-func shuffleTablets(tablets []discovery.TabletStats) {
-	index := 0
+func shuffleTablets(cell string, tablets []discovery.TabletStats) {
```

Comment:
Need unit tests for this function, to test all code paths.

Reply:
👍 ut added to discoverygateway_test